tests/annotator/annotator.cpp

```diff
@@ -1970,78 +2022,64 @@ TEST(Annotator, autoIdOnOutOfDateBuild)
 
     model->component("component2")->setEncapsulationId("changed");
-    EXPECT_FALSE(annotator->isBuilt());
-    EXPECT_EQ("b4da56", annotator->assignId(libcellml::Annotator::Type::COMPONENT_REF, model->component("component2")));
+    EXPECT_EQ("b4da56", annotator->assignId(model->component("component2"), libcellml::CellMLElement::COMPONENT_REF));
     EXPECT_EQ("b4da56", model->component("component2")->encapsulationId());
 
     libcellml::VariablePair connection = std::make_pair(
         model->component("component2")->variable("variable1"),
         model->component("component2")->variable("variable1")->equivalentVariable(0));
     libcellml::Variable::setEquivalenceConnectionId(connection.first, connection.second, "changed");
-    EXPECT_FALSE(annotator->isBuilt());
-    EXPECT_EQ("b4da57", annotator->assignId(libcellml::Annotator::Type::CONNECTION, connection));
+    EXPECT_EQ("b4da57", annotator->assignId(connection));
     EXPECT_EQ("b4da57", libcellml::Variable::equivalenceConnectionId(connection.first, connection.second));
 
     model->setEncapsulationId("changed");
-    EXPECT_FALSE(annotator->isBuilt());
-    EXPECT_EQ("b4da58", annotator->assignId(libcellml::Annotator::Type::ENCAPSULATION, model));
+    EXPECT_EQ("b4da58", annotator->assignId(model, libcellml::CellMLElement::ENCAPSULATION));
     EXPECT_EQ("b4da58", model->encapsulationId());
 
     model->importSource(0)->setId("changed");
-    EXPECT_FALSE(annotator->isBuilt());
-    EXPECT_EQ("b4da59", annotator->assignId(libcellml::Annotator::Type::IMPORT, model->importSource(0)));
+    EXPECT_EQ("b4da59", annotator->assignId(model->importSource(0)));
     EXPECT_EQ("b4da59", model->importSource(0)->id());
 
     libcellml::VariablePair mapping = std::make_pair(
         model->component("component2")->variable("variable2"),
         model->component("component2")->variable("variable2")->equivalentVariable(0));
     libcellml::Variable::setEquivalenceMappingId(mapping.first, mapping.second, "changed");
-    EXPECT_FALSE(annotator->isBuilt());
-    EXPECT_EQ("b4da5a", annotator->assignId(libcellml::Annotator::Type::MAP_VARIABLES, mapping));
+    EXPECT_EQ("b4da5a", annotator->assignId(mapping, libcellml::CellMLElement::MAP_VARIABLES));
     EXPECT_EQ("b4da5a", libcellml::Variable::equivalenceMappingId(mapping.first, mapping.second));
 
     model->setId("changed");
-    EXPECT_FALSE(annotator->isBuilt());
-    EXPECT_EQ("b4da5b", annotator->assignId(libcellml::Annotator::Type::MODEL, model));
+    EXPECT_EQ("b4da5b", annotator->assignId(model));
     EXPECT_EQ("b4da5b", model->id());
 
     model->component("component2")->reset(0)->setId("changed");
-    EXPECT_FALSE(annotator->isBuilt());
-    EXPECT_EQ("b4da5c", annotator->assignId(libcellml::Annotator::Type::RESET, model->component("component2")->reset(0)));
+    EXPECT_EQ("b4da5c", annotator->assignId(model->component("component2")->reset(0), libcellml::CellMLElement::RESET));
```

Comment:
This should/could be with the RESET as default?

Reply:
This is with RESET as default I am testing here that you can pass RESET in with out any problems.